From b561f860f2fefa84459750d576807d214e4aad97 Mon Sep 17 00:00:00 2001
From: Sven Verdoolaege <skimo@kotnet.org>
Date: Sun, 12 Jan 2014 14:35:00 +0100
Subject: [PATCH 04/10] cloog_domain_cube: reimplement using documented
 functions

The original implementation used the undocumented
isl_basic_set_interval function, which will be removed
in the next release of isl.

Signed-off-by: Sven Verdoolaege <skimo@kotnet.org>
Signed-off-by: Cedric Bastoul <cedric.bastoul@unistra.fr>
---
 source/isl/domain.c | 20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

diff --git a/source/isl/domain.c b/source/isl/domain.c
index d11da7b..620584d 100644
--- source/isl/domain.c
+++ source/isl/domain.c
@@ -1389,20 +1389,20 @@ CloogDomain *cloog_domain_cube(CloogState *state,
 				int dim, cloog_int_t min, cloog_int_t max)
 {
 	int i;
-	struct isl_basic_set *cube;
-	struct isl_basic_set *interval;
-	struct isl_basic_set_list *list;
+	isl_space *space;
+	isl_set *cube;
 
 	if (dim == 0)
 		return cloog_domain_universe(state, dim);
 
-	interval = isl_basic_set_interval(state->backend->ctx, min, max);
-	list = isl_basic_set_list_alloc(state->backend->ctx, dim);
-	for (i = 0; i < dim; ++i)
-		list = isl_basic_set_list_add(list, isl_basic_set_copy(interval));
-	isl_basic_set_free(interval);
-	cube = isl_basic_set_list_product(list);
-	return cloog_domain_from_isl_set(isl_set_from_basic_set(cube));
+	space = isl_space_set_alloc(state->backend->ctx, 0, dim);
+	cube = isl_set_universe(space);
+	for (i = 0; i < dim; ++i) {
+		cube = isl_set_lower_bound(cube, isl_dim_set, i, min);
+		cube = isl_set_upper_bound(cube, isl_dim_set, i, max);
+	}
+
+	return cloog_domain_from_isl_set(cube);
 }
 
 
-- 
1.9.3

